src/mlio-py/mlio/contrib/insights/column_analyzer.cc

```diff
@@ -98,6 +98,10 @@ void Column_analyzer::analyze(const mlio::Example &example) const
                     numeric_column_sum += as_float;
                     numeric_column_count++;
 
+                    if (std::fmod(as_float, 1.0) == 0.0) {
```

Comment:
Avoid comparing floating point numbers to fixed values as there might be rounding errors (as in `std::fmod`). There are more sophisticated ways to do fuzzy floating point comparisons, but the trick below should work for this use case:

```c++
// epsilon is a small value (e.g. 1.0e-5f)
if (std::abs(std::round(as_float) - as_float) <= epsilon) {
    stats.numeric_nan_count++;
}
```

Reply:
Thanks! Will use your suggestion